src/algebra/monoid_algebra.lean

```diff
@@ -291,70 +291,90 @@ end algebra
 
 section lift
 
-variables (k G) [comm_semiring k] [monoid G] (A : Type u₃) [semiring A] [algebra k A]
+variables {k G} [comm_semiring k] [monoid G]
+variables {A : Type u₃} [semiring A] [algebra k A]
+variables {B : Type*} [semiring B] [algebra k B]
 local attribute [reducible] monoid_algebra
 
+/--
+Auxiliary definition used to implement lift
+
+For a `monoid_algebra A G`, this is an algebra morphism that applies `H` to the `A` part and `F` to
+the `G` part. Multiplication between `H a` and `F g` must be commutative.
+-/
+def lift_aux (F : G →* B) (H : A →ₐ[k] B) (FH_comm : ∀ a g, H a * F g = F g * H a) : monoid_algebra A G →ₐ[k] B :=
```

Comment:
I suspect this has a lot in common with lemmas like `eval₂_ring_hom' - I intend to attempt to unify them in a follow-up PR.

Reply:
Looks like @urkud beat me to this refactor